adapters/AppLovin/AppLovinAdapter/OpenBidding/GADMAdapterAppLovinRewardedRenderer.m

```diff
@@ -88,73 +80,45 @@ - (instancetype)initWithAdConfiguration:(GADMediationRewardedAdConfiguration *)a
 }
 
 - (void)requestRTBRewardedAd {
-  // Create rewarded video object
-  self.incent = [[ALIncentivizedInterstitialAd alloc] initWithSdk:self.sdk];
   self.isRTBAdRequested = YES;
-
+  self.incent = [[ALIncentivizedInterstitialAd alloc] initWithSdk:self.sdk];
   self.appLovinDelegate = [[GADMAppLovinRewardedDelegate alloc] initWithParentRenderer:self];
   self.incent.adDisplayDelegate = self.appLovinDelegate;
   self.incent.adVideoPlaybackDelegate = self.appLovinDelegate;
-
-  // Load ad
+  
   [self.sdk.adService loadNextAdForAdToken:self.adConfiguration.bidResponse
                                  andNotify:self.appLovinDelegate];
 }
 
 - (void)requestRewardedAd {
-  self.isRTBAdRequested = NO;
-  self.placement =
-      [GADMAdapterAppLovinUtils retrievePlacementFromAdConfiguration:self.adConfiguration];
   self.zoneIdentifier =
       [GADMAdapterAppLovinUtils retrieveZoneIdentifierFromAdConfiguration:self.adConfiguration];
 
-  [GADMAdapterAppLovinUtils log:@"Requesting interstitial for zone: %@ and placement: %@",
-                                self.zoneIdentifier, self.placement];
-
-  // Check if incentivized ad for zone already exists.
-  NSArray *requestedZones;
-  @synchronized(gRequestedRewardedZoneIdentifiers) {
```

Comment:
Gotcha - the issue with this fix was it introduced other bugs. I'll re-write a big portion of it tonight to account for handling of loading RVs of the same zone id at the same time.

Reply:
I don't have data on how often it happens, but we've had publishers complain about behavior when loading multiple ads, so we do try and support this case. I'd like to add this back in.

Re: loading a second ad if the first one timed out - the adapter previously kept a list of "already requested" zone IDs, and removed the lock on an ad unit when an ad failed to load or an ad was shown. However, if there is a timeout, the GMA SDK considers it a failure but the AppLovin SDK considers it a success, and that zone is forever locked. This could be addressed by using dealloc on the renderer to clear the lock on a particular zone, but is not ideal because any bugs in strong references bring this issue back up. Separately, I'm making a proposal to have our SDK notify an adapter when it no longer is waiting for a response (e.g. timeout), so adapters can update state accordingly. That's a long term solution. What I described above is short term.

I'll accept this as is for now.  We are working on a different cleanup CL for the AppLovin adapter that we'll share for your approval, and then I'd like to do 1 more change to add this check back in. We'd be happy to do it or have you do it, either way.

Reply:
> However, if there is a timeout, the GMA SDK considers it a failure but the AppLovin SDK considers it a success, and that zone is forever locked.

Yea, this is what I wanted to fix - `* Fix adapter disallowing future ad loads of a previously-loaded zone that has been timed out by AdMob or not shown by the publisher.`

Since this PR is merged now I guess we can take it in tandem, we can perform the fix after the "different cleanup" change